src/listeners/browser.js

```diff
@@ -1,16 +1,87 @@
+import eventsBulkRequest from '../services/events/bulk';
+import eventsService from '../services/events';
+import impressionsBulkRequest from '../services/impressions/bulk';
+import impressionsService from '../services/impressions';
+import { fromImpressionsCollector } from '../services/impressions/dto';
+import { STORAGE, SETTINGS } from '../utils/context/constants';
+import logFactory from '../utils/logger';
+const log = logFactory('splitio-client:cleanup');
+
 /**
- * Not implemented yet.
+ * We'll listen for 'unload' event over the window object, since it's the standard way to listen page reload and close.
+ *
  */
 export default class BrowserSignalListener {
-  constructor() {
-    // noop
+  constructor(context) {
+    this.storage = context.get(STORAGE);
+    this.settings = context.get(SETTINGS);
+    this._flushEventsAndImpressions = this._flushEventsAndImpressions.bind(this);
+    // this._sendBeacon = this._sendBeacon.bind(this); // not need of binding _sendBeacon
   }
 
+  /**
+   * start method. 
+   * Called when SplitFactory is initialized. 
+   * We add a handler on unload events. The handler flushes remaining impressions and events to the backend.
+   */ 
   start() {
-    // noop
+    log.debug('Registering flush handler when unload page event is triggered.');
+    if (window && window.addEventListener) {
+      window.addEventListener('unload', this._flushEventsAndImpressions);
```

Comment:
make `unload` as a parameter of the class `BrowserSignalListener` in case you subscribe to different event in the future